src/Scripting/ScriptingSystem.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <Library/Logger/LogCategory.h>
+
+#include <string_view>
+#include <string>
+#include <vector>
+#include <unordered_map>
+#include <memory>
+#include <utility>
+#include <sol/sol.hpp>
+
+class DistLogSink;
+class IBindings;
+class PlatformApplication;
+class ScriptLogSink;
+
+class ScriptingSystem {
+ public:
+    ScriptingSystem(std::string_view scriptFolder, std::string_view entryPointFile, PlatformApplication &platformApplication, DistLogSink &distLogSink);
+    ~ScriptingSystem();
+
+    void executeEntryPoint();
+
+    template<typename TBindings, typename ...TArgs>
+    void addBindings(std::string_view bindingTableName, TArgs &&... args) {
+        auto bindings = std::make_unique<TBindings>(std::forward<TArgs>(args) ...);
+        _addBindings(bindingTableName, std::move(bindings));
+    }
+
+    static LogCategory LogCategory;
```

Comment:
@captainurist I'm not sure if the LogCategory for the Scripting should live here or I have to provide a new file instead. Let me know what you think.